idunn/blocks/wikipedia.py

```diff
@@ -3,18 +3,44 @@
 from apistar import validators
 from .base import BaseBlock, BlocksValidator
 from requests.exceptions import HTTPError, RequestException, Timeout
+import pybreaker
 
-def handle_requests_error(f):
-    def wrapper(*args, **kwargs):
-        try:
-            return f(*args, **kwargs)
-        except HTTPError:
-            logging.info("Got HTTP error in {}".format(f.__name__), exc_info=True)
-        except Timeout:
-            logging.warning("External API timed out in {}".format(f.__name__), exc_info=True)
-        except RequestException:
-            logging.error("Got Request exception in {}".format(f.__name__), exc_info=True)
-    return wrapper
+class HTTPError40X(HTTPError):
+    pass
+
+class WikipediaBreaker:
+    _breaker = None
+
+    @classmethod
+    def init_breaker(cls):
+        from app import settings
+        cls._breaker = pybreaker.CircuitBreaker(
+                fail_max = settings['MAXFAIL'],
+                reset_timeout = settings['TIMEOUT'],
+                exclude = [HTTPError40X])
+
+    @classmethod
+    def get_breaker(cls):
+        if cls._breaker is None:
+            cls.init_breaker()
+        return cls._breaker
+
+    @classmethod
+    def handle_requests_error(cls, f):
+        def wrapper(*args, **kwargs):
+            if cls._breaker is None:
```

Comment:
why don't you use `get_breaker`

Reply:
done
(I implemented the method later this is why)